bitcoin/script/script.h

```diff
@@ -587,4 +573,27 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+
+    ([&] (Ts&& input) constexpr {
+        if constexpr (std::is_same_v<std::remove_reference_t<Ts>, CScript>) {
```

Comment:
Just realizing now, this probably needs a `std::remove_cv_t<...>` around the remove_reference as well. Otherwise I think it'll fail to accept references to const CScripts.

Reply:
Yes, tested and you are right.